aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -713,7 +713,12 @@ Tensor zeros_like(const Tensor& self, const TensorOptions& options) {
 
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~ bartlett_window ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Tensor bartlett_window(int64_t window_length, const TensorOptions& options) {
+Tensor bartlett_window(int64_t window_length, c10::optional<c10::ScalarType> dtype, c10::optional<c10::Layout> layout, c10::optional<c10::Device> device, c10::optional<bool> pin_memory) {
```

Comment:
I'm finding it a little weird that:
- In TypeDefault.h, `barlett_window(TensorOptions)` calls `barlett_window(*expanded)`.
- In TensorFactories.cpp, `barlett_window(*expanded)` calls `barlett_window(TensorOptions)`.

It seems to me that the decision for TypeDefault.h is to stop-special-casing TensorOptions in the codegen and the decision in TensorFactories.cpp is because TensorOptions handles the default arguments. However the inconsistency is strange to me.

Reply:
EDIT: Sorry, please ignore. I read the TypeDefault code wrong.